src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -241,29 +243,32 @@ private BoundInterpolatedString BindUnconvertedInterpolatedStringToString(BoundU
             {
                 // Case 1
                 Debug.Assert(unconvertedInterpolatedString.Parts.All(static part => part.Type is null or { SpecialType: SpecialType.System_String }));
-                return constructWithData(BindInterpolatedStringParts(unconvertedInterpolatedString, diagnostics), data: null);
+                return constructWithoutData(BindInterpolatedStringParts(unconvertedInterpolatedString, diagnostics));
             }
 
             // Case 2. Attempt to see if all parts are strings.
-            if (unconvertedInterpolatedString.Parts.Length <= 4 && AllInterpolatedStringPartsAreStrings(unconvertedInterpolatedString.Parts))
-            {
-                return constructWithData(BindInterpolatedStringParts(unconvertedInterpolatedString, diagnostics), data: null);
-            }
-
-            if (tryBindAsHandlerType(out var result))
+            if ((unconvertedInterpolatedString.Parts.Length > 4 || !AllInterpolatedStringPartsAreStrings(unconvertedInterpolatedString.Parts)) &&
+                tryBindAsHandlerType(out var result))
             {
                 // Case 3
                 return result;
             }
 
             // The specifics of 4 vs 5 aren't necessary for this stage of binding. The only thing that matters is that every part needs to be convertible
```

Comment:
Should this comment be updated, since we're now checking whether we can lower to `Concat` or not?